### DISCUSSION TITLE
Finding a page object from url - need a way to find the correct root

### DISCUSSION THREAD

**Discussion by enzedonline** [wagtail/wagtail-localize#361] [Q&A] [answered]
I'm writing a language switcher to work with wagtail-localize. Everything works fine but in it I came across a problem that I can't find a way to get the url_path from a page address - not the best description. 

Example: a site with 4 languages, French is the 4th, it's home page is home-3, so I have the url from the request (eg /fr/nouvelles) and want to get that page object from that url. The url_path for that page will be /home-3/nouvelles, so normal method is to find the root page and swap out the language component for the home page, then grab the page object from Page.objects.get(url_path=url_path).

With wagtail-trans, you could find the root page of the request from Site.find_for_request(request).root_page, but when I do this in wagtail-localize, it always gives me the home page for the default language instead of the home page of the language of the request.

For the moment I've put in a workaround with a dictionary of locales and home pages, but if there is a better way to do this currently, would be good to know.

Maybe something for the ideas list: a method to get the page object from the localized url rather than the url_path

**Comment by enzedonline** [comment]
While the cat was dancing on my face at 5 this morning, the obvious answer to this popped into my head - make sure the home page slugs match the language code that they represent.

I added an override into the home page to make sure this is always the case, and then also any languages added in the future will also conform to this need:

```
    def clean(self, *args, **kwargs):
        self.slug = self.locale.language_code
        super().clean(*args, **kwargs)
```

**Comment by kaedroho** [comment, marked as answer]
Hey @enzedonline 

Sorry for the late response!

The solution to this is to use the `.localized` property on pages. This property finds the translation of the page for the current languages.

For example: ``Site.find_for_request(request).root_page.localized``

**Reply by enzedonline** [reply]
Thank you !
Did a lot of rewriting in the last month, the localized function is extremely useful :)
